### ISSUE TITLE
Named step does not work as parameter inside "then" statement

### ISSUE THREAD

**Issue by zestagio** [mumoshu/variant#122]
Hello,

Previously I created feature request about allow use task result as argument for other tasks (#73). Everything is working fine if I'm not using `if` and `then` statements, but when I place some logic inside `then` statement the task down with error. I prepared simple script to demonstrate it:

```yaml
#!/usr/bin/env var

tasks:
  test1:
    steps:
      - name: foo
        task: foo
      - name: bar
        task: bar
      - task: foobar
        arguments:
          parameter1: "{{ .foo }}"
          parameter2: "{{ .bar }}"

  test2:
    steps:
      - if:
          - task: success
        then:
          - name: foo
            task: foo
          - name: bar
            task: bar
          - task: foobar
            arguments:
              parameter1: "{{ .foo }}"
              parameter2: "{{ .bar }}"

  foobar:
    parameters:
      - name: parameter1
        type: string
      - name: parameter2
        type: string

    script: echo "{{ .parameter1 }}{{ .parameter2 }}"

  success:
    script: exit 0
  foo:
    script: echo foo
  bar:
    script: echo bar
```

Result of executing `test1`:
```
variant do test1
do ≫ starting task foo
foo
do ≫ starting task bar
bar
do ≫ starting task foobar
foobar
```
Result of executing `test2`:
```
variant do test2
do ≫ starting task success
do ≫ starting task foo
foo
do ≫ starting task bar
bar
panic: failed rendering do.test2.argument: template: do.definition.yaml: argument.test2.script:1:3: executing "do.definition.yaml: argument.test2.script" at <.foo>: map has no entry for key "foo"

goroutine 1 [running]:
github.com/mumoshu/variant/pkg.TaskStep.Run.func1(0xc00023a180, 0xa, 0xc0002f45e0, 0xc00006a180)
	/home/circleci/project/pkg/step_task.go:44 +0xeb
github.com/mumoshu/variant/pkg/api/task.Arguments.transformStringValues(0xc000288390, 0xc0001d6c68, 0xc00028f6c0)
	/home/circleci/project/pkg/api/task/arguments.go:80 +0x1ef
github.com/mumoshu/variant/pkg/api/task.Arguments.TransformStringValues(0xc000288390, 0xc0001d6c68, 0x3)
	/home/circleci/project/pkg/api/task/arguments.go:61 +0x39
github.com/mumoshu/variant/pkg.TaskStep.Run(0xc00023a3e8, 0x5, 0xc00023a160, 0x6, 0xc000288390, 0x0, 0x7ffeefbffa50, 0x2, 0x7ffeefbffa50, 0x2, ...)
	/home/circleci/project/pkg/step_task.go:41 +0xad
github.com/mumoshu/variant/pkg.run(0xc00024c200, 0x3, 0x4, 0x7ffeefbffa50, 0x2, 0x7ffeefbffa50, 0x2, 0xc000288f30, 0x0, 0x0, ...)
	/home/circleci/project/pkg/step_if.go:134 +0xac
github.com/mumoshu/variant/pkg.IfStep.Run(0xc00023a378, 0x6, 0xc000237b00, 0x1, 0x1, 0xc00024c200, 0x3, 0x4, 0x1fff548, 0x0, ...)
	/home/circleci/project/pkg/step_if.go:157 +0x10e
github.com/mumoshu/variant/pkg.(*TaskRunner).Run(0xc000265520, 0xc000076300, 0x18bd600, 0x0, 0x0, 0x0, 0x1, 0x0, 0x18a9e71, 0x4)
	/home/circleci/project/pkg/task_runner.go:122 +0x8d0
github.com/mumoshu/variant/pkg.(*Application).RunTask(0xc000076300, 0xc000264ea0, 0x2, 0x2, 0x1fff548, 0x0, 0x0, 0xc000289710, 0xc000289740, 0xc0001d9800, ...)
	/home/circleci/project/pkg/application.go:302 +0xf60
github.com/mumoshu/variant/pkg.(*Application).Run(0xc000076300, 0xc000264ea0, 0x2, 0x2, 0x1fff548, 0x0, 0x0, 0x0, 0x0)
	/home/circleci/project/pkg/application.go:204 +0x172
github.com/mumoshu/variant/pkg.(*CobraAdapter).GenerateCommand.func1(0xc000290780, 0x1fff548, 0x0, 0x0, 0x0, 0x0)
	/home/circleci/project/pkg/cobra.go:59 +0x66
github.com/spf13/cobra.(*Command).execute(0xc000290780, 0x1fff548, 0x0, 0x0, 0xc000290780, 0x1fff548)
	/go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:762 +0x465
github.com/spf13/cobra.(*Command).ExecuteC(0xc000290280, 0xc00028eda0, 0x1, 0x1)
	/go/pkg/mod/github.com/spf13/cobra@v0.0.3/command.go:852 +0x2ec
github.com/mumoshu/variant/pkg.(*CobraApp).Run(0xc00028ed90, 0xc00001e1d0, 0x1, 0x1, 0x2, 0xc00001e1d0, 0x1)
	/home/circleci/project/pkg/variant.go:26 +0x11d
github.com/mumoshu/variant/cmd.Run(0x7ffeefbffa50, 0x2, 0xc00001e1d0, 0x1, 0x1, 0xc00024a000, 0x7ffeefbffa50, 0x2, 0xc00001e1d0, 0x1, ...)
	/home/circleci/project/cmd/run.go:47 +0xfd
github.com/mumoshu/variant/cmd.Def(0xc00024a000, 0x7ffeefbffa50, 0x2, 0xc00001e1d0, 0x1, 0x1, 0xc00006a180, 0xc000264e20, 0x4, 0x4)
	/home/circleci/project/cmd/run_def.go:24 +0x132
github.com/mumoshu/variant/cmd.Dev()
	/home/circleci/project/cmd/dev.go:61 +0x49f
main.main()
	/home/circleci/project/main.go:8 +0x20
```

**Comment by mumoshu** [comment]
@zestagio Thanks for the detailed report! I've spotted the cause and fixed the issue in https://github.com/mumoshu/variant/commit/7c8e9be686768beac76eddb7cea19b5317aa0f2e.

It has already been released as v0.31.1. Would you mind trying with it?

**Comment by zestagio** [comment]
@mumoshu thanks for your quick fix. It works fine now. Tested on 0.34.0
